### PR TITLE
Add Truffle-Migrate to docker-compose

### PR DESCRIPTION
**Pull request by nemaniarjun**
##### Description
This commit adds a new service to docker-compose which runs `truffle migrate` after the other services have been ran. This prevents errors in resolution of contracts and makes the overall
docker experience smooth and error free.

##### Checklist
- [x] Linter status: 100% pass
- [x] Changes don't break existing behavior
- [x] Tests coverage hasn't decreased

##### Affected core subsystem(s)
Docker

##### Refers/Fixes
Fixes: #283

### CODE REVIEW

**Comment by pelsasser** [comment]
@eswarasai - @perfectmak is gonna take a look at this one for us!

File: docker-compose.yml [hunk 1]
```diff
@@ -33,3 +33,19 @@ services:
       - "3000:3000"
     depends_on:
       - truffle
+      - eth-bridge
+
+  truffle-migrate:
+    build: .
```
Comment (perfectmak) [inline docker-compose.yml]: I would suggest you tag the image of the first `truffle` service above and just re-use the image here so you don't need to rebuild the whole image again. @nemaniarjun

Comment (nemaniarjun) [reply docker-compose.yml]: Great idea! Will add in the next iteration!

File: Dockerfile [hunk 2]
```diff
@@ -4,10 +4,8 @@ WORKDIR /app
 
 RUN npm install -g truffle
 
-COPY package.json .
-RUN npm install
```
[the remark this answers is not in the document]
Comment (nemaniarjun) [reply Dockerfile]: OH! That makes sense! I was wondering why it was this way, and thought I would delete it for review! 
TIL something new! :grin:

**Comment by nemaniarjun** [comment]
> @nemaniarjun After reviewing this solution, It doesn't solve the actual problem because the truffle-migrate service keeps re-running because of the restart policy setup on it. Perhaps you can experiment with other restart policies like on-failure and see how best it can work.
Or if you can find a better way to wait for the eth-bridge to finish initializing before running the truffle migrate command.

I tried with `on-failure` and it doesn't seem to fail even when there is an error in connecting to truffle.
I am out of ideas on how to wait for `eth-bridge`. 
I have left some comments on the original issue which I would love your thoughts on.

**Comment by perfectmak** [comment]
@nemaniarjun good to know you noticed it too. I also left some comments on the original issue.

**Comment by pelsasser** [comment]
Closing this PR as per @perfectmak's comments

**Pull request closed without merging**